npm/ng-packs/packages/core/src/lib/tests/routes.service.spec.ts

```diff
@@ -172,6 +172,64 @@ describe('Routes Service', () => {
     });
   });
 
+  describe('#setSingularizeStatus', () => {
+    it('should allow to duplicate routes when called with false', () => {
+      service.setSingularizeStatus(false);
+
+      service.add(routes);
+
+      const flat = service.flat;
+
+      expect(flat.length).toBe(routes.length);
+    });
+
+    it('should allow to duplicate routes with the same name when called with false', () => {
+      service.setSingularizeStatus(false);
+
+      service.add([...routes, { path: '/foo/bar/test', name: 'bar', parentName: 'foo', order: 2 }]);
+
+      const flat = service.flat;
+
+      expect(flat.length).toBe(routes.length + 1);
+    });
+
+    it('should allow to routes with the same name but different parentName when called with false', () => {
+      service.setSingularizeStatus(false);
+
+      service.add([
+        { path: '/foo/bar', name: 'bar', parentName: 'foo', order: 2 },
+        { path: '/foo/bar', name: 'bar', parentName: 'baz', order: 1 },
+      ]);
+
+      const flat = service.flat;
+
+      expect(flat.length).toBe(2);
+    });
+
+    it('should not allow to duplicate routes when called with true', () => {
+      service.setSingularizeStatus(false);
+
+      service.add(routes);
+
+      service.setSingularizeStatus(true);
```

Comment:
Method's default value is `true` we don't need to pass `true`

Reply:
Yes, but this way is more readable. If you know the method definition, it can look unnecessary, but otherwise, dev can think like "setting the singularize status to what?" 😄